mso/resource_mso_remote_location.go

```diff
@@ -0,0 +1,234 @@
+package mso
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSORemoteLocation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSORemoteLocationCreate,
+		Update: resourceMSORemoteLocationUpdate,
+		Read:   resourceMSORemoteLocationRead,
+		Delete: resourceMSORemoteLocationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMSORemoteLocationImport,
+		},
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"scp",
+					"sftp",
+				}, false),
+			},
+			"hostname": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"path": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  22,
+			},
+			"username": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"password": &schema.Schema{
```

Comment:
Was password intentionally set to computed?

Reply:
Yes is there a reason it shouldn't?

Reply:
It would be good to add **Sensitive: True** here

Reply:
ok added for password, sshkey and passphrase